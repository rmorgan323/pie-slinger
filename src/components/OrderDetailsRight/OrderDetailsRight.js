import React from 'react';
import './OrderDetailsRight.css';
import moment from 'moment';
import cap from '../../helper/cap/cap';
import checkNormal from '../../helper/checkNormal/checkNormal';

const OrderDetailsRight = ({ toppings }) => {

  const displayToppings = (toppingsArray) => {
    return toppingsArray.map((topping, index) => {
      return <p key={index}>{cap(topping.name)}: <span className={checkNormal(topping.amount)}>{cap(topping.amount)}</span></p>
    });
  };

  return (
    <div className="OrderDetailsRight">
      <h3>TOPPINGS</h3>
      <div className="toppings-container">
        {displayToppings(toppings)}
      </div>
    </div>
  );
};

export default OrderDetailsRight;