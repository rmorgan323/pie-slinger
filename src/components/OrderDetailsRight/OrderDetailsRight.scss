@import '../../variables.scss';

.OrderDetailsRight {
  width: 260px;
  padding: 0 30px;

  .toppings-container {
    @include border-left;
  }
}