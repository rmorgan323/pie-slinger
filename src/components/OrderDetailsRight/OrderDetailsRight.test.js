import React from 'react';
import { shallow } from 'enzyme';
import OrderDetailsRight from './OrderDetailsRight';

const mockPizza = {
  toppings: [
    { name: "cheese", amount: "normal" },
    { name: "pepperoni", amount: "extra" },
    { name: "sausage", amount: "extra" }
  ]
};

describe('OrderDetailsRight tests', () => {

  it('should match the snapshot', () => {
    const renderedApp = shallow(
      <OrderDetailsRight 
        toppings={mockPizza.toppings}  
      />
    );
    expect(renderedApp).toMatchSnapshot();
  });
  
});