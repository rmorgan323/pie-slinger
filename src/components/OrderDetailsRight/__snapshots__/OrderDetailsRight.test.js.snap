// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetailsRight tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderDetailsRight
    toppings={
        Array [
            Object {
              "amount": "normal",
              "name": "cheese",
            },
            Object {
              "amount": "extra",
              "name": "pepperoni",
            },
            Object {
              "amount": "extra",
              "name": "sausage",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          TOPPINGS
</h3>,
        <div
          className="toppings-container"
>
          <p>
                    Cheese
                    : 
                    <span
                              className={undefined}
                    >
                              Normal
                    </span>
          </p>
          <p>
                    Pepperoni
                    : 
                    <span
                              className="font-bold"
                    >
                              Extra
                    </span>
          </p>
          <p>
                    Sausage
                    : 
                    <span
                              className="font-bold"
                    >
                              Extra
                    </span>
          </p>
</div>,
      ],
      "className": "OrderDetailsRight",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "TOPPINGS",
        },
        "ref": null,
        "rendered": "TOPPINGS",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Cheese
              : 
              <span
                            className={undefined}
              >
                            Normal
              </span>
</p>,
            <p>
              Pepperoni
              : 
              <span
                            className="font-bold"
              >
                            Extra
              </span>
</p>,
            <p>
              Sausage
              : 
              <span
                            className="font-bold"
              >
                            Extra
              </span>
</p>,
          ],
          "className": "toppings-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Cheese",
                ": ",
                <span
                  className={undefined}
>
                  Normal
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Cheese",
              ": ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Normal",
                  "className": undefined,
                },
                "ref": null,
                "rendered": "Normal",
                "type": "span",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Pepperoni",
                ": ",
                <span
                  className="font-bold"
>
                  Extra
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Pepperoni",
              ": ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Extra",
                  "className": "font-bold",
                },
                "ref": null,
                "rendered": "Extra",
                "type": "span",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Sausage",
                ": ",
                <span
                  className="font-bold"
>
                  Extra
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Sausage",
              ": ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Extra",
                  "className": "font-bold",
                },
                "ref": null,
                "rendered": "Extra",
                "type": "span",
              },
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            TOPPINGS
</h3>,
          <div
            className="toppings-container"
>
            <p>
                        Cheese
                        : 
                        <span
                                    className={undefined}
                        >
                                    Normal
                        </span>
            </p>
            <p>
                        Pepperoni
                        : 
                        <span
                                    className="font-bold"
                        >
                                    Extra
                        </span>
            </p>
            <p>
                        Sausage
                        : 
                        <span
                                    className="font-bold"
                        >
                                    Extra
                        </span>
            </p>
</div>,
        ],
        "className": "OrderDetailsRight",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "TOPPINGS",
          },
          "ref": null,
          "rendered": "TOPPINGS",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Cheese
                : 
                <span
                                className={undefined}
                >
                                Normal
                </span>
</p>,
              <p>
                Pepperoni
                : 
                <span
                                className="font-bold"
                >
                                Extra
                </span>
</p>,
              <p>
                Sausage
                : 
                <span
                                className="font-bold"
                >
                                Extra
                </span>
</p>,
            ],
            "className": "toppings-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Cheese",
                  ": ",
                  <span
                    className={undefined}
>
                    Normal
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Cheese",
                ": ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Normal",
                    "className": undefined,
                  },
                  "ref": null,
                  "rendered": "Normal",
                  "type": "span",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Pepperoni",
                  ": ",
                  <span
                    className="font-bold"
>
                    Extra
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Pepperoni",
                ": ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Extra",
                    "className": "font-bold",
                  },
                  "ref": null,
                  "rendered": "Extra",
                  "type": "span",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Sausage",
                  ": ",
                  <span
                    className="font-bold"
>
                    Extra
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Sausage",
                ": ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Extra",
                    "className": "font-bold",
                  },
                  "ref": null,
                  "rendered": "Extra",
                  "type": "span",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
